pennylane/qchem/convert.py

```diff
@@ -153,17 +158,24 @@ def _openfermion_to_pennylane(qubit_operator, wires=None):
 
     xyz2pauli = {"X": qml.PauliX, "Y": qml.PauliY, "Z": qml.PauliZ}
 
+    def _get_op(term, wires):
+        """A function to compute the PL operator associated with the term string."""
+        if len(term) > 1:
+            if active_new_opmath():
+                return qml.prod(*[xyz2pauli[op[1]](wires=wires[op[0]]) for op in term])
+
+            return Tensor(*[xyz2pauli[op[1]](wires=wires[op[0]]) for op in term])
+
+        if len(term) == 1:
+            return xyz2pauli[term[0][1]](wires=wires[term[0][0]])
+
+        return qml.Identity(wires[0])
+
     coeffs, ops = zip(
         *[
             (
                 coef,
-                qml.operation.Tensor(*[xyz2pauli[q[1]](wires=wires[q[0]]) for q in term])
-                if len(term) > 1
-                else (
-                    xyz2pauli[term[0][1]](wires=wires[term[0][0]])
-                    if len(term) == 1
-                    else qml.Identity(wires[0])
-                )
+                _get_op(term, wires)
                 # example term: ((0,'X'), (2,'Z'), (3,'Y'))
             )
```

Comment:
Now that the operator computation is nicely packaged, you could make it more compact for readability:
```Python
coeffs, ops = zip(
    *[(coef, _get_op(term, wires)) for term, coef in qubit_operator.terms.items()]
    # example term: ((0,'X'), (2,'Z'), (3,'Y'))
)
```

Reply:
Added 👍🏼